log.go

```diff
@@ -0,0 +1,59 @@
+package marker
+
+import (
+	"io"
+	"os"
+
+	"github.com/fatih/color"
+)
+
+// StdoutMarkerOption is functional option type for StdoutMarker
+type StdoutMarkerOption func(*StdoutMarker)
+
+// MarkRule contains marking information to be applied on log stream
+type MarkRule struct {
+	Matcher MatcherFunc
+	Color   *color.Color
+}
+
+// StdoutMarker contains specified rules for applying them on output
+type StdoutMarker struct {
+	rules []MarkRule
+	out   io.Writer
+}
+
+// NewStdoutMarker creates a StdoutMarker with default out as os.Stdout
+func NewStdoutMarker(options ...StdoutMarkerOption) *StdoutMarker {
+	logMarker := &StdoutMarker{out: os.Stdout}
+	for _, option := range options {
+		option(logMarker)
+	}
+	return logMarker
+}
+
+// AddRule appends a rule to StdoutMarker and returns itself
+func (s *StdoutMarker) AddRule(rule MarkRule) *StdoutMarker {
+	s.rules = append(s.rules, rule)
+	return s
+}
+
+// AddRules appends all rules from given slice to StdoutMarker rules
+func (s *StdoutMarker) AddRules(rules []MarkRule) {
+	s.rules = append(s.rules, rules...)
+}
+
+// Write marks the text with specified rules and writes the output to specifed out
+func (s StdoutMarker) Write(p []byte) (n int, err error) {
+	marked := string(p)
+	for _, rule := range s.rules {
+		marked = Mark(marked, rule.Matcher, rule.Color)
+	}
+	return s.out.Write([]byte(marked))
```

Comment:
thoughts on a convenience `Write` method that takes in a string?

Reply:
What do you mean by that? 
```golang
type Writer interface {
	Write(p []byte) (n int, err error)
}
```
The `io.Write` takes `[]byte`, if we want to be a `Writer` we need to play the game by the rules :smile:

Reply:
We can extend that interface right? I'm still learning some of the golang internals but my understanding was we had to match the interface but could add additional functionality on top of it. So I was suggesting a `WriteString` func which takes a string, transforms it into a byte array then calls `Write` but if this is bad practice I'm happy to skip!

Reply:
Why should we do that? Most probably user will not deal with the `Write` function, it is just for integrating with `log.Logger`. He/she just can create `StdoutMarker` instance, add the rules and pass it to the `log.Logger` of Golang as Output. (_log.Logger accepts output stream type of **io.Writer**_). 
Then Golang's logger output will be redirected to us and we will mark it before sending it to the os.Stdout!

Simply:
```golang
stdoutMarker := marker.NewStdoutMarker()
stdoutMarker.AddRules(...)
logger := log.New(stdoutMarker, "", 0) // We can pass stdoutMarker as output, since we implemented io.Writer

logger.Println(...) // All
logger.Print(...) // These
logger.Sprintf(...) // are Marked!
// no need to use stdoutMarker.Write(..)
```   
 :tada: 

Since user will not deal with `Write` stuff, i believe there is no need for `WriteString`(unless we are integrating with something that accepts such interface) to prettify our API. Correct me if I get it wrong.

Reply:
Makes sense!